executor/runtime/docker/docker_isolate.go

```diff
@@ -12,15 +12,43 @@ import (
 
 const (
 	titusIsolateHost = "localhost:7500"
+	retryTime        = 100 * time.Millisecond
 )
 
 func waitForTitusIsolate(parentCtx context.Context, taskID string, timeout time.Duration) {
 	waitForTitusIsolateWithHost(parentCtx, taskID, titusIsolateHost, timeout)
 }
 
-func waitForTitusIsolateWithHost(parentCtx context.Context, taskID, host string, timeout time.Duration) {
+func waitForTitusIsolateWithHost(parentCtx context.Context, taskID, host string, timeout time.Duration) bool {
 	ctx, cancel := context.WithTimeout(parentCtx, timeout)
 	defer cancel()
+
+	ticker := time.NewTicker(retryTime)
+
+	firstTry := make(chan struct{}, 1)
+	firstTry <- struct{}{}
+	for {
+		select {
+		case <-ctx.Done():
+			logrus.WithError(ctx.Err()).Warn("Context completed prior to getting a successful result from titus isolate")
+			return false
+
+		case <-ticker.C:
+			if workloadIsolated(ctx, taskID, host) {
+				return true
+			}
+		case <-firstTry:
+			if workloadIsolated(ctx, taskID, host) {
+				return true
+			}
+		}
+	}
+}
+
+func workloadIsolated(ctx context.Context, taskID, host string) bool {
```

Comment:
Minor nitpick: could you rename to something like `isolateWorkload()` (or the like)?  `workloadIsolated` implies that you're checking that the workload was isolated at some time in the past, rather than actively telling titus-isolate to do the isolation.

Reply:
We aren't telling the titus-isolate to do the isolation. I added a comment . accordingly.